fuzz/fuzz_targets/full_stack_target.rs

```diff
@@ -108,12 +108,16 @@ impl BroadcasterInterface for TestBroadcaster {
 #[derive(Clone, PartialEq, Eq, Hash)]
 struct Peer {
 	id: u8,
+	connected: bool,
 }
 impl SocketDescriptor for Peer {
 	fn send_data(&mut self, data: &Vec<u8>, write_offset: usize, _resume_read: bool) -> usize {
 		assert!(write_offset < data.len());
 		data.len() - write_offset
 	}
+	fn disconnect_socket(&mut self) {
+		self.connected = false;
+	}
```

Comment:
Sorry didn't get you on this one, disconnect_socket is used by the network provider to notify that we want to disconnect the peer, right ? So what do we want here is setting a bool in SocketDescriptor object and let the network provider disconnect peer, block further disconnect_event and remove peer infos. Why do we need RefCell ?